rust/agama-network/src/dbus/tree_manager.rs

```diff
@@ -0,0 +1,164 @@
+use agama_lib::error::ServiceError;
+use parking_lot::Mutex;
+use uuid::Uuid;
+
+use crate::{dbus::interfaces, model::*, NetworkSystem};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+/// Handle the objects in the D-Bus tree for the network state
+pub struct Tree {
+    connection: zbus::Connection,
+    network: Arc<Mutex<NetworkSystem>>,
```

Comment:
Ugh, I will need some intro to synchronization techniques, https://docs.rs/parking_lot/0.12.1/parking_lot/type.Mutex.html does not help much

Reply:
I would recommend having a look to:

* [Shared-state concurrency](https://doc.rust-lang.org/book/ch16-03-shared-state.html) as it is the main reason to use mutexes/channels.
* The documentation for the [Mutex from the std](https://doc.rust-lang.org/std/sync/struct.Mutex.html).
* The documentation about [channels](https://doc.rust-lang.org/std/sync/mpsc/fn.channel.html).